src/client/eppo-client.ts

```diff
@@ -987,6 +995,34 @@ export default class EppoClient {
     validateNotBlank(subjectKey, 'Invalid argument: subjectKey cannot be blank');
     validateNotBlank(flagKey, 'Invalid argument: flagKey cannot be blank');
 
+    // Check for override early
+    const overrideVariation = this.overridesStore?.get(getMD5Hash(flagKey));
```

Comment:
security: storing override flag keys as md5 actually removes a layer of security because attacker doesn't need to know the original flag key and md5 hashes are readily available in configuration

performance: hashing is one of the main performance drains when evaluating, so not reusing the hashed flag key will slow down evaluation for the dominant non-overridden flow